flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1435,8 +1437,18 @@ private boolean transitionState(
                         getAttemptId(),
                         currentState,
                         targetState);
-            } else {
-                if (LOG.isInfoEnabled()) {
+            } else if (LOG.isInfoEnabled()) {
+                Optional<NoResourceAvailableException> noResourceException =
+                        findThrowable(error, NoResourceAvailableException.class);
+                if (noResourceException.isPresent()) {
+                    LOG.info(
+                            "{} ({}) switched from {} to {}: {}",
+                            getVertex().getTaskNameWithSubtaskIndex(),
+                            getAttemptId(),
+                            currentState,
+                            targetState,
+                            noResourceException.get().getMessage());
```

Comment:
I think there are `NoResourceAvailableException` that contain a cause that might be helpful to see. Look for `NoResourceAvailableException(String, Throwable)` calls.

Reply:
Hmm..you are right. What do you think about not filling the stacktrace in DeclarativeSlotPoolBridge.failPendingRequests?
(added in c684094020717b1db64e414e645a3939d3aab838)